src/main.rs

```diff
@@ -63,6 +74,216 @@ async fn init_maker(
     }
 }
 
+// TODO: consider introducing components for this (also balance updates below),
+//  Rough idea: Extend MidMarketRate (or introduce new struct) to give us an async next function, hide the fact that it is using channels behind the next function of the component.
+#[allow(clippy::type_complexity)]
+fn init_rate_updates(
+    update_interval_secs: u64,
+) -> (
+    Pin<Box<dyn Future<Output = comit::Never> + Send>>,
+    Receiver<anyhow::Result<MidMarketRate>>,
+) {
+    let (mut sender, receiver) =
+        futures::channel::mpsc::channel::<anyhow::Result<MidMarketRate>>(5);
+
+    let future = async move {
+        loop {
+            let rate = get_btc_dai_mid_market_rate().await;
+
+            if let Err(e) = sender.try_send(rate) {
+                tracing::trace!("Error when sending rate update from sender to receiver.")
+            }
+
+            Delay::new(Duration::from_secs(update_interval_secs)).await;
+        }
+    };
+
+    (Box::pin(future), receiver)
+}
+
+#[allow(clippy::type_complexity)]
+fn init_bitcoin_balance_updates(
+    update_interval_secs: u64,
+    wallet: bitcoin_wallet::Wallet,
+) -> (
+    Pin<Box<dyn Future<Output = comit::Never> + Send>>,
+    Receiver<anyhow::Result<bitcoin::Amount>>,
+) {
+    let (mut sender, receiver) =
+        futures::channel::mpsc::channel::<anyhow::Result<bitcoin::Amount>>(5);
+
+    let future = async move {
+        loop {
+            let balance = wallet.balance().await;
+
+            if let Err(e) = sender.try_send(balance) {
+                tracing::trace!("Error when sending rate update from sender to receiver.")
+            }
+
+            Delay::new(Duration::from_secs(update_interval_secs)).await;
+        }
+    };
+
+    (Box::pin(future), receiver)
+}
+
+// TODO: turn this into an update of DAI and Ether and keep track of the Ether balance in the maker as well
```

Comment:
📣 Why would you want to track the Ether balance in the maker?

Reply:
Because it will be part of the bookkeeping - it is part of reserving funds, because it is necessary for the settlement (i.e. swap execution) and it has to be part of considering to create a new order to ensure the order can be executed.

Reply:
I don't agree on the fact that it is part of book keeping. For the sake of finishing the project we should assume there is always enough gas to execute.

Reply:
OK for me for now, once https://github.com/coblox/nectar/pull/42 is in this logic makes sense again. I'll keep this comment as `Unresolved` for now, to keep this in mind.

Reply:
I removed the TODO from code, will adapt the TODO on basecamp (keep track that tracking Ether for fees is missing so we can potentially build it into a follow up).